src/adapter/src/optimize.rs

```diff
@@ -359,6 +359,22 @@ fn optimize_mir_local(
     Ok::<_, OptimizerError>(expr)
 }
 
+/// This calls a simple optimizer that is able to fully optimize constant INSERTs. For more details,
+/// see [mz_transform::Optimizer::constant_insert_optimizer].
+#[mz_ore::instrument(target = "optimizer", level = "debug", name = "constant_insert")]
+fn optimize_mir_constant_insert(
```

Comment:
Why only `INSERT` statements, rather than any constant view? I think we'll want to remove the `_insert`, as it is called on all expressions, not just inserts. Which is great, btw, just .. confusing to call out inserts as what's going on here.

Reply:
Also, what does it mean to "fully optimize" constant expressions? It might be a moment to spell it out as clearly as possible and make sure we do that. E.g. "any expression all of whose leaves are constants, of size small enough to be inlined and folded" or something.

Reply:
> Why only INSERT statements, rather than any constant view? I think we'll want to remove the _insert, as it is called on all expressions, not just inserts. Which is great, btw, just .. confusing to call out inserts as what's going on here.

Ok, made it more general: it now calls `fold_constants_fixpoint()`, which generally works for constants.

> Also, what does it mean to "fully optimize" constant expressions? It might be a moment to spell it out as clearly as possible and make sure we do that. E.g. "any expression all of whose leaves are constants, of size small enough to be inlined and folded" or something.

I've spelled this out on `fold_constants_fixpoint`, and made some other comments point to that.